File: sample_app/lib/domain/llm_resolver.dart
import 'package:sample_app/models/app_settings.dart';
import 'package:sample_app/domain/llm_usecase.dart';
import 'package:sample_app/data/llm/deepseek_usecase.dart';
import 'package:sample_app/data/llm/yandexgpt_usecase.dart';
import 'package:sample_app/data/llm/tinylama_usecase.dart';

/// Shared resolver that selects concrete LLM use case implementation
/// based on selected neural network in AppSettings.
LlmUseCase resolveLlmUseCase(AppSettings settings) {
  switch (settings.selectedNetwork) {
    case NeuralNetwork.deepseek:
      return DeepSeekUseCase();
    case NeuralNetwork.yandexgpt:
      return YandexGptUseCase();
    case NeuralNetwork.tinylama:
      return TinyLlamaUseCase();
  }
}

File: telegram_monitoring_agent/src/llm_resolver.py
#!/usr/bin/env python3
"""
LLM Resolver - selects appropriate LLM implementation
"""

from .llm_usecase import LlmUseCase
from .deepseek_usecase import DeepSeekUseCase
from .yandexgpt_usecase import YandexGptUseCase


def resolve_llm_usecase(llm_provider: str = "deepseek") -> LlmUseCase:
    """Resolve LLM use case based on provider name"""
    if llm_provider.lower() == "deepseek":
        return DeepSeekUseCase()
    elif llm_provider.lower() == "yandexgpt":
        return YandexGptUseCase()
    else:
        # Default to DeepSeek
        return DeepSeekUseCase()
